core/tx_pool.go

```diff
@@ -1503,9 +1504,22 @@ func (pool *TxPool) executableTxFilter(costLimit *big.Int) func(tx *types.Transa
 // equal number for all for accounts with many pending transactions.
 func (pool *TxPool) truncatePending() {
 	pending := uint64(0)
+
+	parent := pool.chain.CurrentBlock()
+	l1BaseFee := fees.GetL1BaseFee(pool.currentState)
+	pendingBaseFee := misc.CalcBaseFee(pool.chainconfig, parent.Header(), l1BaseFee)
+
+	var allRealPending int64
 	for _, list := range pool.pending {
+		for _, tx := range list.txs.items {
+			if tx.GasTipCapIntCmp(pendingBaseFee) > 0 {
+				allRealPending++
+			}
+		}
 		pending += uint64(list.Len())
 	}
+	realPendingGauge.Update(allRealPending)
```

Comment:
miner might be a better place to put this metric, it already does the filtering (in `NewTransactionsByPriceAndNonce`) at the begining of each block. No need to do it here as well and make txpool reorgs more costly.

https://github.com/scroll-tech/go-ethereum/blob/4b85bbcbd8b3e7dfb2ce9562933c72c55f1319da/miner/scroll_worker.go#L513

Reply:
fixed [c5df443](https://github.com/scroll-tech/go-ethereum/pull/983/commits/c5df443e4b640ff6d531f5365f6f3420b2b1d793)